st2common/st2common/validators/api/reactor.py

```diff
@@ -15,7 +15,11 @@
 
 import six
 
+from apscheduler.triggers.cron import CronTrigger
```

Comment:
Our timer impl is coupled to apscheduler.

rulesengine which implements the timer is the only place that even knows about specific of the trigger.

Validate as much as you like in RulesEngine but please don't bring it upstream. Just because `timer` have some special meaning since `st2` implements these does not mean they should get special preference. As much as possible only https://github.com/StackStorm/st2/blob/master/st2reactor/st2reactor/timer/base.py module should know about implementation details of timers. Anything else that is done in a non-generic way is a hack.